src/flint/fmpz_poly.jl

```diff
@@ -514,7 +514,71 @@ function Base.sqrt(x::fmpz_poly; check::Bool=true)
           (Ref{fmpz_poly}, Ref{fmpz_poly}), z, x))
     check && flag == false && error("Not a square in sqrt")
     return z
- end
+end
+
+# internal use
+# false = definitely not a square
+# true  = may be a square
+# assumes x is not zero
+function issquare_heuristic(x::fmpz_poly)
```

Comment:
This is fine, but I don't think we need issquare_heuristic because fmpz_poly_sqrt already does _fmpz_poly_sqrt_KS, which is written in C and already does a lot of heuristics.

Reply:
I guess I could remove all but the interpolation. On the other hand, the other heuristics ought to be done first. Maybe I still need the check for even degree and zero.

It probably wasn't the best plan to put the heuristics in one of the innermost sqrt functions.

Reply:
fmpz_poly_sqrt does all of the heuristics first

Reply:
Yeah I suppose I should just remove them. I don't know that we do the interpolation heuristic there, but probably what we have is good enough.